packages/docusaurus-theme-classic/src/theme/Navbar/index.js

```diff
@@ -63,7 +64,11 @@ function Navbar() {
   }, [setSidebarShown]);
 
   const onToggleChange = useCallback(
-    e => setTheme(e.target.checked ? 'dark' : ''),
+    e => {
+      const newTheme = e.target.checked ? 'dark' : '';
+      setTheme(newTheme);
+      dispatch('docusaurus-change-theme', {newTheme});
```

Comment:
Why is this `dispatch()` even needed? The `useTheme` hook stores the theme and when it gets updated, them components which are using the value will also re-render. `setTheme` will already trigger an update for those components using `useTheme`. I think this is unnecessary and as a result, we probably don't need the event bus.

Reply:
>  The useTheme hook stores the theme and when it gets updated, them components which are using the value will also re-render.

I tried go that way, but this method does not work for me.

Reply:
I'll patch and try it. Give me a few days, I'm moving countries today 😅

Reply:
Yes please! Although it seems to me that this is not such a bad decision, but with hooks/context it not worked out for me ;(

Reply:
@yangshun any news?

Reply:
> The `useTheme` hook stores the theme and when it gets updated, them components which are using the value will also re-render.

I was completely wrong (wonky memory). `useTheme` is state and local to a component. `setTheme` only updates the theme value within the component.